_pytest/main.py

```diff
@@ -649,36 +645,29 @@ def _recurse(self, path):
         return True
 
     def _tryconvertpyarg(self, x):
-        mod = None
-        path = [os.path.abspath('.')] + sys.path
-        for name in x.split('.'):
-            # ignore anything that's not a proper name here
-            # else something like --pyargs will mess up '.'
-            # since imp.find_module will actually sometimes work for it
-            # but it's supposed to be considered a filesystem path
-            # not a package
-            if name_re.match(name) is None:
-                return x
-            try:
-                fd, mod, type_ = imp.find_module(name, path)
-            except ImportError:
-                return x
-            else:
-                if fd is not None:
-                    fd.close()
+        """Convert a dotted module name to path.
 
-            if type_[2] != imp.PKG_DIRECTORY:
-                path = [os.path.dirname(mod)]
-            else:
-                path = [mod]
-        return mod
+        """
+        import pkgutil
+        try:
+            loader = pkgutil.find_loader(x)
+        except ImportError:
+            return x
+        if loader is None:
+            return x
+        try:
+            path = loader.get_filename()
+        except:
+            path = loader.modules[x][0].co_filename
```

Comment:
In which situations is this necessary? A think a comment here would help. 

I tried to play a bit with this to see what it supposed to do, but didn't obtain much success:

```
Python 3.5.0 (v3.5.0:374f501f4567, Sep 13 2015, 02:16:59) [MSC v.1900 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import pkgutil
>>> loader = pkgutil.find_loader('py')
>>> loader.modules['py']
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'SourceFileLoader' object has no attribute 'modules'
```